pkg/testmachinery/collector/summary.go

```diff
@@ -97,6 +98,7 @@ func (c *collector) generateSummary(tr *tmv1beta1.Testrun, meta *metadata.Metada
 				stepMetadata.Configuration[elem.Name] = elem.Value
 			}
 		}
+		stepMetadata.Annotations[common.LabelTestrunExecutionGroup] = meta.Testrun.ExecutionGroup
```

Comment:
I think the execution group can be empty.
Do we then want to have a execution group annotation that is empty?

Reply:
I think it can't be empty:
https://github.com/gardener/test-infra/blob/1ca8531272ffe13ed280a3c52009650c042c6809/pkg/testrunner/run.go#L31-L33 and 
https://github.com/gardener/test-infra/blob/ea986ed8f24ab178ebc9dfacfb4e3b32f5ba5785/pkg/testmachinery/metadata/metadata.go#L87
are the only setters.
But sure, it doesn't harm to check here as well, should I?

Reply:
With the new testrunner template flags I also added the possibility to not create a execution group.
Although, we do not use that flag together with the summary collector I would check it.
As you said, it does not harm

Reply:
Makes sense in any case, you're right.